src/core/tritonserver.cc

```diff
@@ -2395,8 +2418,9 @@ TRITONSERVER_ServerInferAsync(
     ltrace->SetModelName(lrequest->ModelName());
     ltrace->SetModelVersion(lrequest->ActualModelVersion());
 
-    std::unique_ptr<ni::InferenceTrace> utrace(ltrace);
-    lrequest->SetTrace(std::move(utrace));
+    ni::InferenceTraceProxy* trace_proxy = new ni::InferenceTraceProxy(ltrace);
+    std::shared_ptr<ni::InferenceTraceProxy> strace_proxy(trace_proxy);
+    lrequest->SetTrace(strace_proxy);
```

Comment:
I believe you can replace these 3 lines with:
lrequest->SetTrace(std::make_shared<ni::InferenceTraceProxy>(ltrace))